content/en/blog/_posts/2021-12-15-pod-security-admission-beta.md

```diff
@@ -0,0 +1,774 @@
+
+There is no runtime configurable API for this resource and a cluster administrator would need to specify a path to the file below via the `--admission-control-config-file` flag on the API server. 
+
+In the following resource we are enforcing the baseline policy and warning and auditing the baseline policy. We are also making the kube-system namespace exempt from this policy.
+
+It's not recommended to alter control plane / clusters after install, so let's build a new cluster with a default policy on all namespaces.
+
+First, delete the current cluster.
+
+```shell
+kind delete cluster
+```
+
+Create a Pod Security configuration. 
+
+<!-- TODO: link to the official docs after posted: https://github.com/kubernetes/website/blob/dev-1.23/content/en/docs/tasks/configure-pod-container/enforce-standards-admission-controller.md?plain=1#L25-L55
+--->
+
+```shell
+cat <<EOF > pod-security.yaml
+apiVersion: apiserver.config.k8s.io/v1
+kind: AdmissionConfiguration
+plugins:
+- name: PodSecurity
+  configuration:
+    apiVersion: pod-security.admission.config.k8s.io/v1beta1
+    kind: PodSecurityConfiguration
+    defaults:
+      enforce: "baseline"
+      enforce-version: "latest"
+      audit: "baseline"
+      audit-version: "latest"
+      warn: "restricted"
+      warn-version: "latest"
+    exemptions:
+      # Array of authenticated usernames to exempt.
+      usernames: []
+      # Array of runtime class names to exempt.
+      runtimeClasses: []
+      # Array of namespaces to exempt.
+      namespaces: [kube-system]
+EOF
+```
+
+We now have a default baseline policy. Next pass it to the kind configuration to enable the `--admission-control-config-file` API server argument and pass the policy file. To pass a file to a kind cluster, use a configuration file to pass additional setup instructions. Kind uses `kubeadm` to provision the cluster and the configuration file has the ability to pass `kubeadmConfigPatches` for further customization. In our case, the local file is mounted into the control plane node as `/etc/kubernetes/policies/pod-security.yaml` which is then mounted into the `apiServer` container. We also pass the `--admission-control-config-file` argument pointing to the policy's location.
+
+```shell
+cat <<EOF > kind-config.yaml
+kind: Cluster
+apiVersion: kind.x-k8s.io/v1alpha4
+nodes:
+- role: control-plane
+  kubeadmConfigPatches:
+  - |
+    kind: ClusterConfiguration
+    apiServer:
+        # enable admission-control-config flag on the API server
+        extraArgs:
+          admission-control-config-file: /etc/kubernetes/policies/pod-security.yaml
+        # mount new file / directories on the control plane
+        extraVolumes:
+          - name: policies
+            hostPath: /etc/kubernetes/policies
+            mountPath: /etc/kubernetes/policies
+            readOnly: true
+            pathType: "DirectoryOrCreate"
+  # mount the local file on the control plane
+  extraMounts:
+  - hostPath: ./pod-security.yaml
```

Comment:
Maybe use `/tmp` directory for consistency with tutorial: https://github.com/kubernetes/website/blob/004e9a48b3cb7912c85a24f1fcb6812f91abb688/content/en/docs/tutorials/pod-security/cluster-level-pss.md#configure-the-api-server and easier clean up?

Reply:
IIRC `/tmp` works but, given the permissions that /tmp usually has, you're mounting a very open dir into the containers for potential abuse. While I think it's unlikely, it's probably not a great practice if someone adopts it in production. I like the idea of taking the current directory for the users interpretation.

I don't think either of the tutorials need to change but I could be sold on unifying. (maybe it's a good idea to have diverging examples?)